### ISSUE TITLE
Haskell `SPC m h` bindings not working

### ISSUE THREAD

**Issue by tko** [syl20bnr/spacemacs#3173]
1. With haskell layer (and shm mode) enabled open a `test.hs` file with following contents:
    ```haskell
import Control.Monad (when)

main = when True $ putStrLn "Hello"
```
2. Move the cursor to the latter `when`
3. Press <kbd>SPC m h t</kbd> or <kbd>SPC m h i</kbd>

Expected result: identifier/type information as documented

Actual result: Error: `<interactive>:1:1-4: Not in scope: ‘when’`

(This is after executing `haskell-session-change` as suggested in the first error when trying these keybindings.)

**Comment by tko** [comment]
<kbd>SPC m h T</kbd> is similar

It looks like those key bindings aren't operating on the code that's in the buffer, but rather the code that's in the repl running in the background, e.g. previously sent with <kbd>SPC m s b</kbd> or something.

I must be misunderstanding ghc-mod and/or how the packages are interacting; I have `haskell-enable-ghc-mod-support` enabled and from what I recall from vim ghc-mod is able to give the type info without any extra steps.

**Comment by tko** [comment]
There's another PR #3199 to bind the ghc-mod functions to even shorter prefix.

**Comment by d12frosted** [comment]
Wehn you enable `interactive-haskell-mode` - those bindings require associated REPL. If you disable it, but still have `ghc-mod` - `C-c C-t` will be bound to `ghc-show-type` function.

**Comment by d12frosted** [comment]
@syl20bnr this is fixed by #3345 :smile:
